packages/native/types/base.d.ts

```diff
@@ -89,21 +87,26 @@ export interface StyledOptions<Props> {
   shouldForwardProp?(propName: PropertyKey): boolean
 }
 
+export interface StyledWithComponent<ComponentProps extends {}> {
+  withComponent<
+    Component extends ComponentType<ComponentPropsWithoutRef<Component>>
+  >(
+    component: Component
+  ): StyledComponent<ComponentProps & ComponentPropsWithoutRef<Component>>
+  withComponent<ComponentName extends ReactNativeComponentNames>(
+    component: ReactNativeComponents[ComponentName]
+  ): StyledComponent<ComponentProps, ReactNativeComponentProps<ComponentName>>
+}
```

Comment:
these `withComponent` signatures were pulled out into their own interface so that we can use a type union below for `StyledComponent`.

Reply:
wouldnt we be able to do the same without pulling out into its own interface?

Reply:
I think you're right, we don't need this to be its own interface any more (i think it was useful before when i needed to use a union).